EIPS/eip-7377.md

```diff
@@ -0,0 +1,127 @@
+
+One significant problems that hasn't been resolved though is what to do about the EOA users of *today*. Manually migrating assets from an EOA is not a tenable path to converting Ethereum's user base to smart contract wallets. We must provide a mechanism, embedded in the protocol, to migrate. This EIP proposes such mechanism.
+
+## Specification
+
+At the fork block `X`, introduce the MigrationTransaction transaction type.
+
+### MigrationTransaction
+
+#### Definition
+
+| field                  | type      |
+|------------------------|-----------|
+| `chainId`              | `int256`  |
+| `nonce`                | `uint64`  |
+| `maxFeePerGas`         | `int256`  |
+| `maxPriorityFeePerGas` | `int256`  |
+| `gasLimit`             | `uint64`  |
+| `codePtr`              | `address` |
+| `storage`              | `List[Tuple[uint256, uint256]]` |
+| `data`                 | `bytes`   |
+| `value`                | `int256`  |
+| `accessList`           | `List[Tuple[address, List[uint256]]]` |
+| `yParity`              | `uint8`   |
+| `v`                    | `uint256` |
+| `r`                    | `uint256` |
+
+The EIP-2718 `TransactionType` is `0x04` and the `TransactionPayload` is `rlp([chainId, nonce, maxFeePerGas, maxPriorityFeePerGas, gasLimit, codePtr, storageTuples, data, value, accessList, yParity, r, s])`.
+
+The transaction's signature hash is `keccak256(0x04 || rlp([chainId, nonce, maxFeePerGas, maxPriorityFeePerGas, gasLimit, codePtr, storageTuples, data, value, accessList])`
+
+#### Validation
+
+A MigrationTransaction is considered valid if the follow properties hold:
+
+* all [EIP-1559](./eip-1559.md) properties, unless specified otherwise
+* the code at `codePtr` is less than the [EIP-170](./eip-170) limit of `24576`
+* the code at `codePtr` must not have size `0`
+
+The intrinsic gas calculation modified from [EIP-1559](./eip-1559) to be `21000 + 16 * non-zero calldata bytes + 4 * zero calldata bytes + 1900 * access list storage key count + 2400 * access list address count + 15000 * length of storage`.
+
+#### Processing
+
+Execution a MigrationTransaction transaction has two parts.
+
+##### Contract Deployment
+
+Unlike standard contract deployment, the MigrationTransaction directly specifies what `code` value of the sender's account should be set to.
+
+As the first step of processing the transaction, set the sender's `code` value equal to `state[tx.codePtr].code`. Next, iterate through each tuple in `tx.storage` and in the sender's storage trie, set `storage[t.first] = t.second`.
+
+##### Transaction Execution
+
+Now instantiate an EVM call into the sender's account using the same rules as [EIP-1559](./eip-1559) and set the transaction's origin to be `keccak256(sender)[0..20]`.
```

Comment:
```suggestion
Now instantiate an EVM call into the sender's account using the same rules as EIP-1559 and set the transaction's origin to be `keccak256(sender)[0..20]`.
```
```suggestion
Now instantiate an EVM call into the sender's account using the same rules as [EIP-1559](./eip-1559.md) and set the transaction's origin to be `keccak256(sender)[0..20]`.
```
I don't know why this is being done with EIP-1559 specifically. Pick one, but if there is a re-link, then the .md extension is not optional.